python/cudf/cudf/core/index.py

```diff
@@ -454,15 +454,28 @@ def isin(self, values, level=None):
 
         """
 
-        if isinstance(values, Index):
-            values = column.as_column(values)
```

Comment:
We don't need to have `values` to be converted to a `Column` yet as in `Column.isin` we are now directly trying to create a typecasted column with a `try/catch` block.